recipes/android-ndk/all/conanfile.py

```diff
@@ -263,7 +263,7 @@ def package_info(self):
             self.env_info.READELF = self._define_tool_var("READELF", "readelf", True)
             # there doesn't seem to be an 'elfedit' included anymore.
         else:
-            self.env_info.LD = self._define_tool_var("LD", "ld")
+            self.env_info.LD = self._define_tool_var("LD", "ld", skip_check=True)
```

Comment:
Disabling a failing check will result in a bugged package because `LD` will be incorrect.
In https://github.com/conan-io/conan-center-index/issues/7953#issuecomment-962421225, you wrote that `ld` is missing, but there exists a bfd and gold linker.

So I propose the following fix:
add a new method (or modify the existing one) to check for different suffices.
e.g.
```python
self.env_info.LD = self._define_tool_var("LD", "ld", possible_suffixes=["", ".bfd", ".gold"])
```
`_define_tool_var` should then try all sufixes and use the first successful one.

Reply:
I see your point, but I'm not sure if this will result in the correct setting. My aim here was to "restore" the old behavior for a recently introduced check (which is not working here). I don't know much about Android NDK, and I hope the original author of this part of code knows more about. 

My way of thinking was: `LD` is not a path to a tool, it is only the " path stem", which will be extended by the caller with the appropriate suffix. But after reviewing this idea multiple times I'm less convinced ;)

I played around a little bit with the `gtest` example, and finally I recognized that none of this env variables is used. Because of CMake I'm sure. So this are settings for other, low-level build scripts. 

I only found one reference inside the NDK package:



```makefile
# build/tools/common-build-host-funcs.sh line 697ff

bh_setup_host_env ()
{
    CC=$BH_HOST_CONFIG-gcc
    CXX=$BH_HOST_CONFIG-g++
    LD=$BH_HOST_CONFIG-ld
    AR=$BH_HOST_CONFIG-ar
    AS=$BH_HOST_CONFIG-as
    RANLIB=$BH_HOST_CONFIG-ranlib
    NM=$BH_HOST_CONFIG-nm
    STRIP=$BH_HOST_CONFIG-strip
    STRINGS=$BH_HOST_CONFIG-strings
    export CC CXX LD AR AS RANLIB NM STRIP STRINGS
...
```

Maybe this is the source for this part of code.

OK, in short: I can do that, but I'm not sure if it is the right thing, and which one of the two (bld or gold) should be taken? And for other platforms, are there other suffixes/defaults?

Maybe someone with more knowledge of NDK internals can give a clue.

Reply:
True, it is mostly autotools-type + custom build scripts that are of concern here, CMake uses CXX/CC as the linker.

You detected a real bug here: an environment variable with an invalid value.
So I feel uncomfortable ignoring it.

On the other hand, I think it would be better to just not define `LD`.
As I've written above, most build systems use `CC/CXX` as default linker executable.

This would also avoid a lot of warnings/errors causes by build scripts passing compiler flags to the linker.
e.g. binutil's linkers doesn't understand `-D` macro definitions (they don't make sense when linking)

So please just remove the `LD` definitions, and add a comment instead.
Something like:
```
# don't define LD to $prefix-ld/$prefix-ld.gold/$prefix-bfd. Build scripts should use the compiler (CC/CXX) as frontend.
```